src/dotty/tools/dotc/transform/LambdaLift.scala

```diff
@@ -102,9 +102,10 @@ class LambdaLift extends MiniPhase with IdentityDenotTransformer { thisTransform
 
     /** Mark symbol `sym` as being free in `enclosure`, unless `sym`
      *  is defined in `enclosure` or there is a class between `enclosure`s owner
-     *  and the owner of `sym`.
-     *  Return `true` if there is no class between `enclosure` and
-     *  the owner of sym.
+     *  and the owner of `sym`. Also, update lifted owner of `enclosure` so
```

Comment:
> there is a class between `enclosure`s owner and the owner of `sym`

It is still no clear that method follows the contract. The method does not reffer to the `sym.owner` at all, instead it uses `sym.enclosingClass`.

I believe that this is an important method which needs a more verbose documentation, both for the contract and for the implementation